ReactWindows/ReactNative/Accessibility/AccessibilityHelper.cs

```diff
@@ -170,431 +313,289 @@ public static void SetAccessibilityTraits(IAccessible view, object accessibility
-                }
-                SetChildrenAccessibilityView(childPeer, accessibilityView);
-            }
-        }
+#if PERF_LOG
+            s_treeContext.Value.ProcessedNodesCount = 0;
+            var savedDirtyNodesCount = s_treeContext.Value.DirtyNodesCount;
+#endif
 
-        /// <summary>
-        /// Returns true if in <paramref name="elementPeer"/> parents chain there is an UIElement with
-        /// ImportantForAccessibility value of Yes or NoHideDescendants or (Auto and AccessibilityLabelAttached attached property is set).
-        /// 'Yes' or (Auto and AccessibilityLabelAttached attached property is set) denotes that the element is visible
-        /// to the narrator as a container i.e. its children are 'hidden'. 'NoHideDescendants' denotes that both
-        /// the element and its children are 'hidden'. 'Hidden' children have AccessibilityView set to 'Raw'.
-        /// </summary>
-        /// <param name="elementPeer"></param>
-        /// <returns></returns>
-        private static bool IsHiddenByAncestor(AutomationPeer elementPeer)
-        {
-            if (elementPeer == null)
-            {
-                return false;
-            }
-
-            bool result;
-            switch (elementPeer.Navigate(AutomationNavigationDirection.Parent))
+            // Recurse through all known roots
+            foreach (var root in s_treeContext.Value.RootViews)
             {
-                case FrameworkElementAutomationPeer parentFEAutomationPeer:
-                    if (parentFEAutomationPeer.Owner != null)
-                    {
-                        if (DoesHideDescendants(parentFEAutomationPeer.Owner))
-                        {
-                            return true;
-                        }
-                    }
-                    result = IsHiddenByAncestor(parentFEAutomationPeer);
-                    break;
-                case ItemAutomationPeer parentIAutomationPeer:
-                    if (parentIAutomationPeer.Item is UIElement parentUIElement)
-                    {
-                        if (DoesHideDescendants(parentUIElement))
-                        {
-                            return true;
-                        }
-                    }
-                    result = IsHiddenByAncestor(parentIAutomationPeer);
-                    break;
-                case AutomationPeer parentAutomationPeer:
-                    result = IsHiddenByAncestor(parentAutomationPeer);
-                    break;
-                case null:
-                    result = false;
-                    break;
-                default:
-                    result = false;
-                    break;
+                UpdateAccessibilityPropertiesForTree(root);
             }
 
-            return result;
+            // Not dirty anymore
+            s_treeContext.Value.Dirty = false;
+
+#if PERF_LOG
+            RnLog.Info("AccessibilityHelper", $"Stats: ElementCount: {s_treeContext.Value.ElementCount}, " +
```

Comment:
>RnLog [](start = 12, length = 5)

RnLog isn't a thing yet, right? Even if it's #ifdef'd out, we probably shouldn't commit it.

Reply:
Oopsie, I wanted to say Debug.WriteLine...

---
In reply to: [177185658](https://github.com/Microsoft/react-native-windows/pull/1733#discussion_r177185658) [](ancestors = 177185658)